src/lib/VirtualScroller.js

```diff
@@ -166,34 +186,139 @@ class VirtualScroller {
         }
 
-        let numItemsRendered = 0;
+        let newStartOffset = offset;
+        let newEndOffset = offset + this.maxRenderedItems;
+        // If the default count of items to render exceeds the totalItems count
+        // then just render up to the end
+        if (newEndOffset >= this.totalItems) {
+            newEndOffset = this.totalItems - 1;
+        }
 
         // Create a new list element to be swapped out for the existing one
         const newListEl = this.createListElement();
-        while (numItemsRendered < count) {
-            const rowEl = this.renderItem(offset + numItemsRendered);
-            newListEl.appendChild(rowEl);
-            numItemsRendered += 1;
+        const children = Array.prototype.slice.call(this.listEl.children);
+
+        if (curStartOffset <= offset && offset <= curEndOffset) {
+            // Scenario #1: New start offset falls within the current range of items rendered
+            //      |--------------------|
+            //  curStartOffset       curEndOffset
+            //          |--------------------|
+            //   newStartOffset          newEndOffset
+            newStartOffset = curEndOffset + 1;
+            // clone elements from newStartOffset to curEndOffset
+            this.cloneItems(newListEl, children, offset - curStartOffset, curEndOffset - curStartOffset);
+            // then create elements from curEnd + 1 to newEndOffset
+            this.createItems(newListEl, newStartOffset, newEndOffset);
+        } else if (curStartOffset <= newEndOffset && newEndOffset <= curEndOffset) {
+            // Scenario #2: New end offset falls within the current range of items rendered
+            //                |--------------------|
+            //          curStartOffset        curEndOffset
+            //          |--------------------|
+            //    newStartOffset        newEndOffset
+
+            // create elements from newStartOffset to curStart - 1
+            this.createItems(newListEl, offset, curStartOffset - 1);
+            // then clone elements from curStartOffset to newEndOffset
+            this.cloneItems(newListEl, children, 0, newEndOffset - curStartOffset);
+        } else {
+            // Scenario #3: New range has no overlap with current range of items
+            //                          |--------------------|
+            //                    curStartOffset        curEndOffset
+            //  |--------------------|
+            // newStartOffset    newEndOffset
+            this.createItems(newListEl, newStartOffset, newEndOffset);
         }
 
         this.scrollingEl.replaceChild(newListEl, this.listEl);
         this.listEl = newListEl;
     }
 
+    /**
+     * Clones a subset of the HTMLElements from the oldList to the newList.
+     * The newList element is modified.
+     *
+     * @param {HTMLElement} newList - the new `li` element
+     * @param {Array<HTMLElement>} oldList - the children of the old `li` element
```

Comment:
Minor, but it seems inconsistent to have `newList` represent a parent `ol`, but `oldList` is the children of the old `ol`. It also looks like the comments may need to be corrected from `li` -> `ol`.